### PR TITLE
caching list webjobs call fixes #1855

### LINKED ISSUE #1855: Call to Api/WebJobs/ takes more than 20 seconds sometimes leading to the GeoMaster timing out and Ibiza WebJobs blade failing to load, need to improve performance

### PR DESCRIPTION
**Pull request by shobak101**

### CODE REVIEW

File: Kudu.Core/Jobs/TriggeredJobsScheduler.cs [hunk 1]
```diff
@@ -94,6 +94,7 @@ private void OnJobChanged(string jobName)
                 triggeredJobSchedule.Dispose();
                 _triggeredJobsSchedules.Remove(jobName);
             }
+            ((TriggeredJobsManager) _triggeredJobsManager).ListJobRefreshNeeded = true;
```
Comment (davidebbo) [inline Kudu.Core/Jobs/TriggeredJobsScheduler.cs]: Won't that break any unit test that mocks the interface?

File: Kudu.Core/Jobs/ContinuousJobsManager.cs [hunk 2]
```diff
@@ -18,6 +18,19 @@ public class ContinuousJobsManager : JobsManagerBase<ContinuousJob>, IContinuous
     {
         private const string StatusFilesSearchPattern = ContinuousJobStatus.FileNamePrefix + "*";
         private const string Localhost = "127.0.0.1";
+        public bool ListJobRefreshNeeded
```
Comment (davidebbo) [inline Kudu.Core/Jobs/ContinuousJobsManager.cs]: Use auto-prop

File: Kudu.Core/Jobs/ContinuousJobsManager.cs [hunk 3]
```diff
@@ -18,6 +18,19 @@ public class ContinuousJobsManager : JobsManagerBase<ContinuousJob>, IContinuous
...
+            get
+            {
+                return _listJobRefreshNeeded;
+            }
+            set
+            {
+                _listJobRefreshNeeded = value;
+            }
+        }
+        private bool _listJobRefreshNeeded = true;
```
Comment (ahmelsayed) [inline Kudu.Core/Jobs/ContinuousJobsManager.cs]: What is this used for? and why have a backing field?

File: Kudu.Core/Jobs/ContinuousJobsManager.cs (hunk 2 above)
Comment (davidebbo) [inline Kudu.Core/Jobs/ContinuousJobsManager.cs]: Also, why not just set the cache to null when we need a refresh and do away with this separate prop?

File: Kudu.Core/Jobs/TriggeredJobsManager.cs [hunk 4]
```diff
@@ -31,7 +44,14 @@ public TriggeredJobsManager(ITraceFactory traceFactory, IEnvironment environment
 
         public override IEnumerable<TriggeredJob> ListJobs()
         {
-            return ListJobsInternal();
+            if (TriggeredJobCache == null || ListJobRefreshNeeded)
```
Comment (suwatch) [inline Kudu.Core/Jobs/TriggeredJobsManager.cs]: var triggerJobCache = TriggerJobCache;
...
return triggerJobCache;

File: Kudu.Core/Jobs/TriggeredJobsScheduler.cs (hunk 1 above)
Comment (suwatch) [inline Kudu.Core/Jobs/TriggeredJobsScheduler.cs]: this could be just set cache to null.

Comment (suwatch) [inline Kudu.Core/Jobs/TriggeredJobsScheduler.cs]: The cache object should simply be static class member, no?

**Pull request closed without merging**